File: app/src/main/java/com/prinkal/pos/app/dialog/ColoredSimpleProgressDialog.java
package com.prinkal.pos.app.dialog;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.res.ColorStateList;
import android.graphics.Color;
import android.graphics.PorterDuff;
import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import androidx.appcompat.widget.LinearLayoutCompat;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.prinkal.pos.app.R;
import com.prinkal.pos.app.databinding.DialogProgressColoredBinding;

import androidx.annotation.ColorRes;
import androidx.annotation.DimenRes;
import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.core.content.ContextCompat;

public class ColoredSimpleProgressDialog extends Dialog {
  private final DialogProgressColoredBinding binding;
  
  private Context mContext;
  
  @ColorRes
  private int mBackGroundColorResource;
  
  @ColorRes
  private int mProgressBarColorResource;
  
  private Drawable mBackGroundDrawable;
  
  private Drawable mProgressBarDrawable;
  
  private Drawable mSeparatorDrawable;
  
  private String mTextViewText;
  
  private String mTitleText;
  
  private int mTextViewTextSize;
  
  private int mTitleTextSize;
  
  private int TEXT_VIEW_SCALE_TYPE = 0;
  
  private int TITLE_TEXT_VIEW_SCALE_TYPE = 0;
  
  private Drawable mTextBackgroundDrawable;
  
  private boolean mShowTitle;
  
  private boolean mShowTitleSeparator = true;
  
  public ColoredSimpleProgressDialog(@NonNull Context context) {
    this(context, false, (OnCancelListener)null);
  }
  
  public ColoredSimpleProgressDialog(@NonNull Context context, boolean cancelable, @Nullable DialogInterface.OnCancelListener cancelListener) {
    super(context, cancelable, cancelListener);
    requestWindowFeature(1);
    this.mContext = context;
    this.binding = DialogProgressColoredBinding.inflate(LayoutInflater.from(context), null, false);
    setContentView(this.binding.getRoot());
  }
  
  public ColoredSimpleProgressDialog setBackGroundColor(@ColorRes int color) {
    this.mBackGroundColorResource = color;
    return this;
  }
  
  public ColoredSimpleProgressDialog setBackGroundColor(Color color) {
    this.mBackGroundDrawable = (Drawable)new ColorDrawable(Color.parseColor(color.toString()));
    return this;
  }
  
  public ColoredSimpleProgressDialog setBackGroundColor(String color) {
    this.mBackGroundDrawable = (Drawable)new ColorDrawable(Color.parseColor(color));
    return this;
  }
  
  public ColoredSimpleProgressDialog setBackGroundDrawable(@DrawableRes int drawableInt) {
    this.mBackGroundDrawable = ContextCompat.getDrawable(this.mContext, drawableInt);
    return this;
  }
  
  public ColoredSimpleProgressDialog setBackGroundDrawable(Drawable drawable) {
    this.mBackGroundDrawable = drawable;
    return this;
  }
  
  public ColoredSimpleProgressDialog setProgressBarColor(@ColorRes int color) {
    this.mProgressBarColorResource = color;
    return this;
  }
  
  public ColoredSimpleProgressDialog setProgressBarColor(Color color) {
    this.mProgressBarDrawable = (Drawable)new ColorDrawable(Color.parseColor(color.toString()));
    return this;
  }
  
  public ColoredSimpleProgressDialog setProgressBarColor(String color) {
    this.mProgressBarDrawable = (Drawable)new ColorDrawable(Color.parseColor(color));
    return this;
  }
  
  public ColoredSimpleProgressDialog setProgressBarDrawable(Drawable drawable) {
    this.mProgressBarDrawable = drawable;
    return this;
  }
  
  public ColoredSimpleProgressDialog setText(String text) {
    this.mTextViewText = text;
    return this;
  }
  
  public ColoredSimpleProgressDialog setText(@StringRes int text) {
    this.mTextViewText = this.mContext.getString(text);
    return this;
  }
  
  public ColoredSimpleProgressDialog setTitleText(String text) {
    this.mTitleText = text;
    return this;
  }
  
  public ColoredSimpleProgressDialog setTitleText(@StringRes int text) {
    this.mTitleText = this.mContext.getString(text);
    return this;
  }
  
  public ColoredSimpleProgressDialog setContentTextSize(int size) {
    this.mTextViewTextSize = size;
    return this;
  }
  
  public ColoredSimpleProgressDialog setContentTextSizeInSP(int size) {
    this.TEXT_VIEW_SCALE_TYPE = 2;
    this.mTextViewTextSize = size;
    return this;
  }
  
  public void setContentTextSizeResource(@DimenRes int size) {
    this.mTextViewTextSize = this.mContext.getResources().getDimensionPixelSize(size);
  }
  
  public ColoredSimpleProgressDialog setTitleTextSize(int size) {
    this.mTitleTextSize = size;
    return this;
  }
  
  public ColoredSimpleProgressDialog setTitleTextSizeWithUnit(int size, int unit) {
    switch (unit) {
      case 1:
        this.TITLE_TEXT_VIEW_SCALE_TYPE = 1;
        this.mTitleTextSize = size;
        return this;
      case 2:
        this.TITLE_TEXT_VIEW_SCALE_TYPE = 2;
        this.mTitleTextSize = size;
        return this;
      case 3:
        this.TITLE_TEXT_VIEW_SCALE_TYPE = 3;
        this.mTitleTextSize = size;
        return this;
      case 4:
        this.TITLE_TEXT_VIEW_SCALE_TYPE = 4;
        this.mTitleTextSize = size;
        return this;
      case 5:
        this.TITLE_TEXT_VIEW_SCALE_TYPE = 5;
        this.mTitleTextSize = size;
        return this;
    } 
    this.TITLE_TEXT_VIEW_SCALE_TYPE = 0;
    this.mTitleTextSize = size;
    return this;
  }
  
  public void setTitleTextSizeResource(@DimenRes int size) {
    this.mTitleTextSize = this.mContext.getResources().getDimensionPixelSize(size);
  }
  
  public ColoredSimpleProgressDialog setTextBackGroundColor(Color color) {
    this.mTextBackgroundDrawable = (Drawable)new ColorDrawable(Color.parseColor(color.toString()));
    return this;
  }
  
  public ColoredSimpleProgressDialog setTextBackGroundColor(String color) {
    this.mTextBackgroundDrawable = (Drawable)new ColorDrawable(Color.parseColor(color));
    return this;
  }
  
  public ColoredSimpleProgressDialog setTextBackGroundDrawable(@DrawableRes int drawableInt) {
    this.mTextBackgroundDrawable = ContextCompat.getDrawable(this.mContext, drawableInt);
    return this;
  }
  
  public ColoredSimpleProgressDialog setTextBackGroundDrawable(Drawable drawable) {
    this.mTextBackgroundDrawable = drawable;
    return this;
  }
  
  public ColoredSimpleProgressDialog showTitle(boolean show) {
    this.mShowTitle = show;
    return this;
  }
  
  public ColoredSimpleProgressDialog showTitleSeparator(boolean show) {
    this.mShowTitleSeparator = show;
    return this;
  }
  
  public ColoredSimpleProgressDialog setTitleAndContentSeparatorColor(Color color) {
    this.mSeparatorDrawable = (Drawable)new ColorDrawable(Color.parseColor(color.toString()));
    return this;
  }
  
  public ColoredSimpleProgressDialog setTitleAndContentSeparatorColor(String color) {
    this.mSeparatorDrawable = (Drawable)new ColorDrawable(Color.parseColor(color));
    return this;
  }
  
  public ColoredSimpleProgressDialog setTitleAndContentSeparatorColor(@ColorRes int color) {
    this.mSeparatorDrawable = (Drawable)new ColorDrawable(ContextCompat.getColor(this.mContext, color));
    return this;
  }
  
  public ProgressBar getProgressBar() {
    return this.binding.progressBar;
  }
  
  public TextView getContentView() {
    return this.binding.text;
  }
  
  public TextView getTitleView() {
    return this.binding.titleTextView;
  }
  
  public View getTitleSeparatorView() {
    return this.binding.titleAndContentSeparator;
  }
  
  public LinearLayoutCompat getTitleContainerview() {
    return this.binding.titleContainer;
  }
  
  public LinearLayoutCompat getContentContainerView() {
    return this.binding.contentContainer;
  }
  
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    if (this.mBackGroundColorResource != 0)
      this.binding.window.setBackgroundColor(ContextCompat.getColor(this.mContext, this.mBackGroundColorResource)); 
    if (this.mBackGroundDrawable != null)
      if (Build.VERSION.SDK_INT >= 16) {
        this.binding.window.setBackground(this.mBackGroundDrawable);
      } else {
        this.binding.window.setBackgroundDrawable(this.mBackGroundDrawable);
      }  
    if (this.mTextViewText != null)
      this.binding.text.setText(this.mTextViewText); 
    if (this.mTitleText != null) {
      this.binding.titleTextView.setText(this.mTitleText);
      this.mShowTitle = true;
    } 
    if (this.mShowTitle) {
      this.binding.titleContainer.setVisibility(View.VISIBLE);
    } else {
      this.binding.titleContainer.setVisibility(View.GONE);
    } 
    if (this.mShowTitleSeparator) {
      this.binding.titleAndContentSeparator.setVisibility(View.VISIBLE);
    } else {
      this.binding.titleAndContentSeparator.setVisibility(View.GONE);
    } 
    if (this.mTextViewTextSize != 0)
      this.binding.text.setTextSize(this.TEXT_VIEW_SCALE_TYPE, this.mTextViewTextSize); 
    if (this.mTitleTextSize != 0)
      this.binding.titleTextView.setTextSize(this.TITLE_TEXT_VIEW_SCALE_TYPE, this.mTitleTextSize); 
    if (this.mTextBackgroundDrawable != null)
      if (Build.VERSION.SDK_INT >= 16) {
        this.binding.text.setBackground(this.mTextBackgroundDrawable);
      } else {
        this.binding.text.setBackgroundDrawable(this.mTextBackgroundDrawable);
      }  
    if (this.mSeparatorDrawable != null)
      if (Build.VERSION.SDK_INT >= 16) {
        this.binding.titleAndContentSeparator.setBackground(this.mSeparatorDrawable);
      } else {
        this.binding.titleAndContentSeparator.setBackgroundDrawable(this.mSeparatorDrawable);
      }  
    boolean indeterminate = this.binding.progressBar.isIndeterminate();
    try {
      if (this.mProgressBarColorResource != 0)
        if (Build.VERSION.SDK_INT >= 23) {
          if (indeterminate) {
            this.binding.progressBar.setIndeterminateTintList(ColorStateList.valueOf(this.mContext.getResources().getColor(this.mProgressBarColorResource, null)));
          } else {
            this.binding.progressBar.setProgressTintList(ColorStateList.valueOf(this.mContext.getResources().getColor(this.mProgressBarColorResource, null)));
          } 
        } else if (Build.VERSION.SDK_INT >= 21) {
          if (indeterminate) {
            this.binding.progressBar.setIndeterminateTintList(ColorStateList.valueOf(this.mContext.getResources().getColor(this.mProgressBarColorResource)));
          } else {
            this.binding.progressBar.setProgressTintList(ColorStateList.valueOf(this.mContext.getResources().getColor(this.mProgressBarColorResource)));
          } 
        } else if (indeterminate) {
          this.binding.progressBar.getIndeterminateDrawable().setColorFilter(this.mContext.getResources().getColor(this.mProgressBarColorResource), PorterDuff.Mode.SRC_IN);
        } else {
          this.binding.progressBar.getProgressDrawable().setColorFilter(this.mContext.getResources().getColor(this.mProgressBarColorResource), PorterDuff.Mode.SRC_IN);
        }  
      if (this.mProgressBarDrawable != null)
        if (this.mProgressBarDrawable instanceof ColorDrawable) {
          if (indeterminate) {
            this.binding.progressBar.getIndeterminateDrawable().setColorFilter(((ColorDrawable)this.mProgressBarDrawable).getColor(), PorterDuff.Mode.SRC_IN);
          } else {
            this.binding.progressBar.getProgressDrawable().setColorFilter(((ColorDrawable)this.mProgressBarDrawable).getColor(), PorterDuff.Mode.SRC_IN);
          } 
        } else if (indeterminate) {
          this.binding.progressBar.setIndeterminateDrawable(this.mProgressBarDrawable);
        } else {
          this.binding.progressBar.setProgressDrawable(this.mProgressBarDrawable);
        }  
    } catch (Exception e) {
      e.printStackTrace();
    } 
  }
  
  public static ColoredSimpleProgressDialog getIntstantProgressDialog(Context context) {
    ColoredSimpleProgressDialog coloredSimpleProgressDialog = new ColoredSimpleProgressDialog(context);
    coloredSimpleProgressDialog.showTitle(true);
    coloredSimpleProgressDialog.showTitleSeparator(true);
    coloredSimpleProgressDialog.setTitleText(R.string.progress_dialog_title_text);
    coloredSimpleProgressDialog.setText(R.string.progress_dialog_content_text);
    coloredSimpleProgressDialog.setCanceledOnTouchOutside(true);
    coloredSimpleProgressDialog.setCancelable(true);
    return coloredSimpleProgressDialog;
  }
}

File: app/src/main/java/com/prinkal/pos/app/dialog/ColoredSimpleProgressDialog.kt
package com.prinkal.pos.app.dialog

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.PorterDuff
import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.Drawable
import android.os.Build
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import androidx.annotation.ColorRes
import androidx.annotation.DimenRes
import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import androidx.appcompat.widget.LinearLayoutCompat
import androidx.core.content.ContextCompat
import com.prinkal.pos.app.R
import com.prinkal.pos.app.databinding.DialogProgressColoredBinding

class ColoredSimpleProgressDialog @JvmOverloads constructor(context: Context, cancelable: Boolean = false, cancelListener: DialogInterface.OnCancelListener? = null as DialogInterface.OnCancelListener?) : Dialog(context, cancelable, cancelListener) {
    private val binding: DialogProgressColoredBinding
    private val mContext: Context

    @ColorRes
    private var mBackGroundColorResource = 0

    @ColorRes
    private var mProgressBarColorResource = 0
    private var mBackGroundDrawable: Drawable? = null
    private var mProgressBarDrawable: Drawable? = null
    private var mSeparatorDrawable: Drawable? = null
    private var mTextViewText: String? = null
    private var mTitleText: String? = null
    private var mTextViewTextSize = 0
    private var mTitleTextSize = 0
    private var TEXT_VIEW_SCALE_TYPE = 0
    private var TITLE_TEXT_VIEW_SCALE_TYPE = 0
    private var mTextBackgroundDrawable: Drawable? = null
    private var mShowTitle = false
    private var mShowTitleSeparator = true
    fun setBackGroundColor(@ColorRes color: Int): ColoredSimpleProgressDialog {
        mBackGroundColorResource = color
        return this
    }

    fun setBackGroundColor(color: Color): ColoredSimpleProgressDialog {
        mBackGroundDrawable = ColorDrawable(Color.parseColor(color.toString()))
        return this
    }

    fun setBackGroundColor(color: String?): ColoredSimpleProgressDialog {
        mBackGroundDrawable = ColorDrawable(Color.parseColor(color))
        return this
    }

    fun setBackGroundDrawable(@DrawableRes drawableInt: Int): ColoredSimpleProgressDialog {
        mBackGroundDrawable = ContextCompat.getDrawable(mContext, drawableInt)
        return this
    }

    fun setBackGroundDrawable(drawable: Drawable?): ColoredSimpleProgressDialog {
        mBackGroundDrawable = drawable
        return this
    }

    fun setProgressBarColor(@ColorRes color: Int): ColoredSimpleProgressDialog {
        mProgressBarColorResource = color
        return this
    }

    fun setProgressBarColor(color: Color): ColoredSimpleProgressDialog {
        mProgressBarDrawable = ColorDrawable(Color.parseColor(color.toString()))
        return this
    }

    fun setProgressBarColor(color: String?): ColoredSimpleProgressDialog {
        mProgressBarDrawable = ColorDrawable(Color.parseColor(color))
        return this
    }

    fun setProgressBarDrawable(drawable: Drawable?): ColoredSimpleProgressDialog {
        mProgressBarDrawable = drawable
        return this
    }

    fun setText(text: String?): ColoredSimpleProgressDialog {
        mTextViewText = text
        return this
    }

    fun setText(@StringRes text: Int): ColoredSimpleProgressDialog {
        mTextViewText = mContext.getString(text)
        return this
    }

    fun setTitleText(text: String?): ColoredSimpleProgressDialog {
        mTitleText = text
        return this
    }

    fun setTitleText(@StringRes text: Int): ColoredSimpleProgressDialog {
        mTitleText = mContext.getString(text)
        return this
    }

    fun setContentTextSize(size: Int): ColoredSimpleProgressDialog {
        mTextViewTextSize = size
        return this
    }

    fun setContentTextSizeInSP(size: Int): ColoredSimpleProgressDialog {
        TEXT_VIEW_SCALE_TYPE = 2
        mTextViewTextSize = size
        return this
    }

    fun setContentTextSizeResource(@DimenRes size: Int) {
        mTextViewTextSize = mContext.resources.getDimensionPixelSize(size)
    }

    fun setTitleTextSize(size: Int): ColoredSimpleProgressDialog {
        mTitleTextSize = size
        return this
    }

    fun setTitleTextSizeWithUnit(size: Int, unit: Int): ColoredSimpleProgressDialog {
        when (unit) {
            1 -> {
                TITLE_TEXT_VIEW_SCALE_TYPE = 1
                mTitleTextSize = size
                return this
            }
            2 -> {
                TITLE_TEXT_VIEW_SCALE_TYPE = 2
                mTitleTextSize = size
                return this
            }
            3 -> {
                TITLE_TEXT_VIEW_SCALE_TYPE = 3
                mTitleTextSize = size
                return this
            }
            4 -> {
                TITLE_TEXT_VIEW_SCALE_TYPE = 4
                mTitleTextSize = size
                return this
            }
            5 -> {
                TITLE_TEXT_VIEW_SCALE_TYPE = 5
                mTitleTextSize = size
                return this
            }
        }
        TITLE_TEXT_VIEW_SCALE_TYPE = 0
        mTitleTextSize = size
        return this
    }

    fun setTitleTextSizeResource(@DimenRes size: Int) {
        mTitleTextSize = mContext.resources.getDimensionPixelSize(size)
    }

    fun setTextBackGroundColor(color: Color): ColoredSimpleProgressDialog {
        mTextBackgroundDrawable = ColorDrawable(Color.parseColor(color.toString()))
        return this
    }

    fun setTextBackGroundColor(color: String?): ColoredSimpleProgressDialog {
        mTextBackgroundDrawable = ColorDrawable(Color.parseColor(color))
        return this
    }

    fun setTextBackGroundDrawable(@DrawableRes drawableInt: Int): ColoredSimpleProgressDialog {
        mTextBackgroundDrawable = ContextCompat.getDrawable(mContext, drawableInt)
        return this
    }

    fun setTextBackGroundDrawable(drawable: Drawable?): ColoredSimpleProgressDialog {
        mTextBackgroundDrawable = drawable
        return this
    }

    fun showTitle(show: Boolean): ColoredSimpleProgressDialog {
        mShowTitle = show
        return this
    }

    fun showTitleSeparator(show: Boolean): ColoredSimpleProgressDialog {
        mShowTitleSeparator = show
        return this
    }

    fun setTitleAndContentSeparatorColor(color: Color): ColoredSimpleProgressDialog {
        mSeparatorDrawable = ColorDrawable(Color.parseColor(color.toString()))
        return this
    }

    fun setTitleAndContentSeparatorColor(color: String?): ColoredSimpleProgressDialog {
        mSeparatorDrawable = ColorDrawable(Color.parseColor(color))
        return this
    }

    fun setTitleAndContentSeparatorColor(@ColorRes color: Int): ColoredSimpleProgressDialog {
        mSeparatorDrawable = ColorDrawable(ContextCompat.getColor(mContext, color))
        return this
    }

    val progressBar: ProgressBar
        get() = binding.progressBar
    val contentView: TextView
        get() = binding.text
    val titleView: TextView
        get() = binding.titleTextView
    val titleSeparatorView: View
        get() = binding.titleAndContentSeparator
    val titleContainerview: LinearLayoutCompat
        get() = binding.titleContainer
    val contentContainerView: LinearLayoutCompat
        get() = binding.contentContainer

    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        if (mBackGroundColorResource != 0) binding.window.setBackgroundColor(ContextCompat.getColor(mContext, mBackGroundColorResource))
        if (mBackGroundDrawable != null) if (Build.VERSION.SDK_INT >= 16) {
            binding.window.background = mBackGroundDrawable
        } else {
            binding.window.setBackgroundDrawable(mBackGroundDrawable)
        }
        if (mTextViewText != null) binding.text.text = mTextViewText
        if (mTitleText != null) {
            binding.titleTextView.text = mTitleText
            mShowTitle = true
        }
        if (mShowTitle) {
            binding.titleContainer.visibility = View.VISIBLE
        } else {
            binding.titleContainer.visibility = View.GONE
        }
        if (mShowTitleSeparator) {
            binding.titleAndContentSeparator.visibility = View.VISIBLE
        } else {
            binding.titleAndContentSeparator.visibility = View.GONE
        }
        if (mTextViewTextSize != 0) binding.text.setTextSize(TEXT_VIEW_SCALE_TYPE, mTextViewTextSize.toFloat())
        if (mTitleTextSize != 0) binding.titleTextView.setTextSize(TITLE_TEXT_VIEW_SCALE_TYPE, mTitleTextSize.toFloat())
        if (mTextBackgroundDrawable != null) if (Build.VERSION.SDK_INT >= 16) {
            binding.text.background = mTextBackgroundDrawable
        } else {
            binding.text.setBackgroundDrawable(mTextBackgroundDrawable)
        }
        if (mSeparatorDrawable != null) if (Build.VERSION.SDK_INT >= 16) {
            binding.titleAndContentSeparator.background = mSeparatorDrawable
        } else {
            binding.titleAndContentSeparator.setBackgroundDrawable(mSeparatorDrawable)
        }
        val indeterminate = binding.progressBar.isIndeterminate
        try {
            if (mProgressBarColorResource != 0) if (Build.VERSION.SDK_INT >= 23) {
                if (indeterminate) {
                    binding.progressBar.indeterminateTintList = ColorStateList.valueOf(mContext.resources.getColor(mProgressBarColorResource, null))
                } else {
                    binding.progressBar.progressTintList = ColorStateList.valueOf(mContext.resources.getColor(mProgressBarColorResource, null))
                }
            } else if (Build.VERSION.SDK_INT >= 21) {
                if (indeterminate) {
                    binding.progressBar.indeterminateTintList = ColorStateList.valueOf(mContext.resources.getColor(mProgressBarColorResource))
                } else {
                    binding.progressBar.progressTintList = ColorStateList.valueOf(mContext.resources.getColor(mProgressBarColorResource))
                }
            } else if (indeterminate) {
                binding.progressBar.indeterminateDrawable.setColorFilter(mContext.resources.getColor(mProgressBarColorResource), PorterDuff.Mode.SRC_IN)
            } else {
                binding.progressBar.progressDrawable.setColorFilter(mContext.resources.getColor(mProgressBarColorResource), PorterDuff.Mode.SRC_IN)
            }
            if (mProgressBarDrawable != null) if (mProgressBarDrawable is ColorDrawable) {
                if (indeterminate) {
                    binding.progressBar.indeterminateDrawable.setColorFilter((mProgressBarDrawable as ColorDrawable?)!!.color, PorterDuff.Mode.SRC_IN)
                } else {
                    binding.progressBar.progressDrawable.setColorFilter((mProgressBarDrawable as ColorDrawable?)!!.color, PorterDuff.Mode.SRC_IN)
                }
            } else if (indeterminate) {
                binding.progressBar.indeterminateDrawable = mProgressBarDrawable
            } else {
                binding.progressBar.progressDrawable = mProgressBarDrawable
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    companion object {
        fun getIntstantProgressDialog(context: Context): ColoredSimpleProgressDialog {
            val coloredSimpleProgressDialog = ColoredSimpleProgressDialog(context)
            coloredSimpleProgressDialog.showTitle(true)
            coloredSimpleProgressDialog.showTitleSeparator(true)
            coloredSimpleProgressDialog.setTitleText(R.string.progress_dialog_title_text)
            coloredSimpleProgressDialog.setText(R.string.progress_dialog_content_text)
            coloredSimpleProgressDialog.setCanceledOnTouchOutside(true)
            coloredSimpleProgressDialog.setCancelable(true)
            return coloredSimpleProgressDialog
        }
    }

    init {
        requestWindowFeature(1)
        mContext = context
        binding = DialogProgressColoredBinding.inflate(LayoutInflater.from(context), null, false)
        setContentView(binding.root)
    }
}